src/Pages/SignUpFormik.tsx

```diff
@@ -29,102 +29,109 @@ const theme = createTheme();
+  //   return;
+  // }
 
-    const lastLogin = 'null';
-    const acceptedTermAndConditions = false;
+  // const lastLogin = 'null';
+  // const acceptedTermAndConditions = false;
 
-    const account = {
-      _id: undefined,
-      emailAddress,
-      isAdmin,
-      password,
-      firstName,
-      lastName,
-      companyName,
-      lastLogin,
-      acceptedTermAndConditions,
-    };
-
-    const validationSchema = yup.object({
-      emailAddress: yup
-        .string()
-        .email('Bitte vollständige E-Mail-Adresse eingeben!')
-        .required('E-Mail-Adresse ist erforderlich.'),
-      name: yup
-        .string()
-        .min(3, 'Name should be of minimum 3 characters length')
-        .required('Name ist erforderlich.'),
-    });
+  // const account = {
+  //   _id: undefined,
+  //   emailAddress,
+  //   isAdmin,
+  //   password,
+  //   firstName,
+  //   lastName,
+  //   companyName,
+  //   lastLogin,
+  //   acceptedTermAndConditions,
+  // };
+  // };
+  const validationSchema = yup.object({
+    emailAddress: yup
+      .string()
+      .email('Bitte vollständige E-Mail-Adresse eingeben!')
+      .required('E-Mail-Adresse ist erforderlich.'),
+    password: yup
+      .string()
+      .min(8, 'Passwort sollte mindestens eine Länge von 8 Zeichen haben.')
+      .required('Passwort ist erforderlich.'),
+    firstName: yup
+      .string()
+      .min(2, 'Vorname sollte mindestens eine Länge von 2 Zeichen haben.')
+      .required('Vorname ist erforderlich.'),
+    lastName: yup
+      .string()
+      .min(2, 'Nachname sollte mindestens eine Länge von 2 Zeichen haben.')
+      .required('Nachname ist erforderlich.'),
+    companyName: yup
+      .string()
+      .required('Der Name des Unternehmens ist erforderlich.'),
+  });
 
-    const formik = useFormik({
-      initialValues: {
-        _id: undefined,
-        emailAddress: '',
-        isAdmin: '',
-        password: '',
-        firstName: '',
-        lastName: '',
-        companyName: '',
-        lastLogin: '',
-        acceptedTermAndConditions: false,
-      },
-      validationSchema: validationSchema,
-      onSubmit: (values) => {
-        CreateAccount(account)
-          .then((account) => {
-            // alert('success');
-            // success
-            setCookie('email', emailAddress, { path: '/' });
-            return navigate('/my-products');
-          })
-          .catch((msg) => {
-            alert('error');
-            // error
-          });
-      },
-    });
-  };
+  const formik = useFormik({
+    initialValues: {
+      _id: undefined,
+      emailAddress: '',
+      isAdmin: '',
+      password: '',
+      firstName: '',
+      lastName: '',
+      companyName: '',
+      lastLogin: '',
+      acceptedTermAndConditions: false,
+    },
+    validationSchema: validationSchema,
+    onSubmit: (values) => {
+      CreateAccount(values)
+        .then((values) => {
+          setCookie('email', emailAddress, { path: '/' });
```

Comment:
`emailAddress` does not exist in this scope. You probably want to get the email from the `values` variable instead.

Reply:
so just `values` instead of `emailAddress`?

Reply:
No. `values` is probably of type `AccountType`, right? You can get the emailAddress field out of it.